wycheproof/sb_wycheproof_hmac_test.c

```diff
@@ -117,18 +117,14 @@ static _Bool sb_test_wycheproof_hmac(const char* const filename)
     SB_TEST_ASSERT(sb_test_open(filename, &tests));
 
     while(parse_hmac_test_block(tests, &key, &msg, &tag, &result)) {
+        sb_hmac_sha256(&hmac, actual_out, key.buf, key.len, 
+                                          msg.buf, msg.len);
         if (result) {
-            sb_hmac_sha256(&hmac, actual_out, key.buf, key.len, 
-                                              msg.buf, msg.len);
-
-            SB_TEST_ASSERT_EQUAL(*tag.buf, *actual_out);
+            // The known tag and the actual output should be equal.
+            SB_TEST_ASSERT_EQUAL_2(1, *tag.buf, *actual_out, tag.len);
```

Comment:
Use `SB_TEST_ASSERT_EQUAL` not `SB_TEST_ASSERT_EQUAL_2`. The macro is magic and will do the right thing automatically.

Reply:
I switched to using SB_TEST_ASSERT_EQUAL_2 because I needed to set the size parameter to the length of the tag. The SB_TEST_ASSERT_EQUAL macro uses sizeof(the second parameter) as the compare condition so I got some test failures for the cases with shorter tags.

Reply:
Makes sense!